test/one/profiler/test/TestResult.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package one.profiler.test;
+
+public class TestResult {
+    private final TestStatus status;
+    private final Throwable throwable;
+    private final SkipReason skipReason;
+
+    public TestResult(TestStatus status) {
+        this(status, (Throwable) null);
+    }
+
+    public TestResult(TestStatus status, Throwable throwable) {
```

Comment:
What about leaving just one private constructor? Outside this class, new instances should be created only via static methods: pass, fail, etc.